src/Form.js

```diff
@@ -1,19 +1,97 @@
 import React from "react";
 import "./Form.css";
+import { useState } from "react";
 
 function Form() {
+  const [userInput, setUserInput] = useState("");
+  const [selection, SetUserSelection] = useState("");
+  //const [err, setErr] = useState("");
+  const [finalResult, setFinalResult] = useState("")
+
+  //let sum = 0;
+  let result;
+  let obj = {};
+
+  const handleUserInput = (e) => {
+    setUserInput(e.target.value);
+    console.log(e.target.value);
+  };
+
+  const handleUserSel = (e) => {
+    SetUserSelection(e.target.value);
+  };
+
+  function mode(arr) {
+    return arr
+      .sort(
+        (a, b) =>
+          arr.filter((v) => v === a).length - arr.filter((v) => v === b).length
+      )
+      .pop();
+  }
+
+  const form = (e) => {
+    e.preventDefault();
+    let inputArr = userInput.split(',')
+    let isValid = true
+    for(let input of inputArr) {
+      if (isNaN(Number(input))) {
+        isValid = false
+        break;
+      }
+    }
+    if (!selection || !userInput || !isValid) {
+      result  = "Invalid input.";
+      
+    }else if (selection === "sum") {
+      result = 0
+     inputArr.forEach((input) => {
+        result += Number(input);
+      });
+      //console.log(sum);
+    }
+    else if (selection === "average") {
+      result = inputArr.reduce((a, b) => a + b, 0) / inputArr.length;
```

Comment:
inputArr is an array of strings representing numbers [ '1', '2' , '3' ]. Just like we did for sum, we need to turn each of these strings into numbers. The best way to do so here is to map over them. 
`result =
        inputArr.map((num) => Number(num)).reduce((a, b) => a + b, 0) /
        inputArr.length;`
Otherwise it can lead to weird results

Reply:
Yes. I will do that now. Thank you Raz !